rocky_worlds_utils/jwst/archive_utils.py

```diff
@@ -0,0 +1,217 @@
+#! /usr/bin/env python
+"""
+JWST Archive utilities module for the Rocky Worlds DDT project. Most of the functions
+here are used to query for targets included in the RWDDT TUC that are not a part of
+our programs.
+
+Authors
+-------
+- Mees Fix <<mfix@stsci.edu>>
+
+Use
+---
+>>> from rocky_worlds_utils.jwst.archive_utils import query_nexsci_archive
+>>> data, pref_idx = query_nexsci_archive("TRAPPIST-1 b")
+"""
+
+from copy import copy
+import warnings
+
+from astroquery.ipac.nexsci.nasa_exoplanet_archive import NasaExoplanetArchive
+from astropy.coordinates import SkyCoord
+from astropy.table import MaskedColumn
+from astropy.time import Time
+import astropy.units as u
+from astroquery.mast import MastMissions
+import numpy as np
+
+
+from rocky_worlds_utils.constants import JWST_PROGRAMS
+
+__all__ = [
+    "check_jwst_observations",
+    "check_jwst_event_type",
+    "query_nexsci_archive",
+]
+
+
+def check_jwst_observations(ra, dec, radius=0.1):
+    """Check MAST to see if target ra and dec has JWST observations.
+
+    Parameters
+    ----------
+    ra : float
+        Target Right Ascension (units: deg)
+    dec : float
+        Target Declination (units: deg)
+    radius : float
+        Search radius around ra and dec [default: 0.1]
+
+    Returns
+    -------
+    results : astropy.Table.table
+        Astropy table of query results
+    """
+    regionCoords = SkyCoord(ra, dec, unit=("deg", "deg"))
+    missions = MastMissions(mission="jwst")
+    results = missions.query_region(
+        regionCoords,
+        radius=radius,
+        select_cols=[
+            "targprop",
+            "targ_ra",
+            "targ_dec",
+            "instrume",
+            "exp_type",
+            "opticalElements",
+            "date_obs",
+            "duration",
+            "program",
+            "observtn",
+            "visit",
+            "pi_name",
+            "proposal_type",
+            "proposal_cycle",
+            "targtype",
+            "access",
+        ],
+    )
+
+    # drop rows that contain observations from RWDDT programs
+    for rwddt_program in JWST_PROGRAMS:
+        if rwddt_program in results["program"]:
+            idx = np.where(results["program"] == rwddt_program)[0]
+            results.remove_rows(idx)
+        else:
+            continue
+
+    results.add_index("ArchiveFileID")
+
+    return results
+
+
+def check_jwst_event_type(period, planet_ephemeris, jwst_observations):
+    """
+    This function tries to figure out, given some target information and
+    an observation start and end time,what exoplanet event is being
+    observed: transit, eclipse, phase curve, or nothing.
+
+    This code may be buggy and has not been well-tested.
```

Comment:
Let's talk to @hdiamondlowe about this comment 😅 @ladsantos doesn't like saying that our code is buggy.